### PR TITLE
partners: Add `langchain-snowflake` package with `CortexSearchRetriever`

### PR DESCRIPTION
**Pull request by sfc-gh-nmoiseyev**
- **Description:** Adds a Snowflake partner integration in a new `langchain-snowflake` package
- **Dependencies:** `snowflake.core >= 0.8.0, <1.0`

### CODE REVIEW

File: libs/partners/snowflake/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+license = "MIT"
+
+[tool.poetry.urls]
+"Source Code" = "https://github.com/langchain-ai/langchain/tree/master/libs/partners/snowflake"
+"Release Notes" = "https://github.com/langchain-ai/langchain/releases?q=tag%3A%22snowflake%3D%3D0%22&expanded=true"
+
+[tool.poetry.dependencies]
+python = ">=3.9,<3.12"
+langchain-core = ">=0.2.40"
+snowflake = ">=0.8.0"
```
Comment (efriis) [inline libs/partners/snowflake/pyproject.toml:17]: should we upper bound this as `<1`?

File: libs/partners/snowflake/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+repository = "https://github.com/langchain-ai/langchain"
+license = "MIT"
+
+[tool.poetry.urls]
+"Source Code" = "https://github.com/langchain-ai/langchain/tree/master/libs/partners/snowflake"
+"Release Notes" = "https://github.com/langchain-ai/langchain/releases?q=tag%3A%22snowflake%3D%3D0%22&expanded=true"
+
+[tool.poetry.dependencies]
+python = ">=3.9,<3.12"
+langchain-core = ">=0.2.40"
```
Comment (efriis) [inline libs/partners/snowflake/pyproject.toml:16]:
```suggestion
langchain-core = ">=0.2.40,<0.4"
```

File: libs/partners/snowflake/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+readme = "README.md"
+repository = "https://github.com/langchain-ai/langchain"
+license = "MIT"
+
+[tool.poetry.urls]
+"Source Code" = "https://github.com/langchain-ai/langchain/tree/master/libs/partners/snowflake"
+"Release Notes" = "https://github.com/langchain-ai/langchain/releases?q=tag%3A%22snowflake%3D%3D0%22&expanded=true"
+
+[tool.poetry.dependencies]
+python = ">=3.9,<3.12"
```
Comment (efriis) [inline libs/partners/snowflake/pyproject.toml:15]: why isn't this compatible with python 3.12?

**Review by efriis — COMMENTED** [review]
will slack about integration testing creds too!

File: libs/partners/snowflake/pyproject.toml (hunk 3 above)
Comment (sfc-gh-nmoiseyev) [reply libs/partners/snowflake/pyproject.toml:15]: unfortunately, a current limitation of the `snowflake.core` package: https://pypi.org/project/snowflake.core/

but checking with the team on why this is the case

File: libs/partners/snowflake/pyproject.toml (hunk 1 above)
Comment (sfc-gh-nmoiseyev) [reply libs/partners/snowflake/pyproject.toml]: Can't hurt. Done

File: libs/partners/snowflake/pyproject.toml (hunk 3 above)
Comment (sfc-gh-nmoiseyev) [reply libs/partners/snowflake/pyproject.toml:15]: double checked and this is due to a dependency on Snowpark in `snowflake.core`. The next version of `snowflake.core` won't have this dependency

**Comment by efriis** [comment]
doing in langchain-ai/langchain-snowflake isntead!

**Pull request closed without merging**